core/src/main/kotlin/gropius/dto/input/OptionalInputExtensions.kt

```diff
@@ -0,0 +1,53 @@
+package gropius.dto.input
+
+import com.expediagroup.graphql.generator.execution.OptionalInput
+import com.expediagroup.graphql.generator.scalars.ID
+import kotlin.reflect.KProperty0
+
+/**
+ * Executes [block] with the value if `this is OptionalInput.Defined`
+ *
+ * @param block executed if input is defined
+ */
+inline fun <T> OptionalInput<T>.ifPresent(block: (T) -> Unit) {
+    if (this is OptionalInput.Defined) {
+        block(this.value!!)
+    }
+}
+
+/**
+ * `true` if the input is present
+ */
+val OptionalInput<*>.isPresent: Boolean get() = this is OptionalInput.Defined
+
+/**
+ * If present, returns its value, otherwise [value]
+ *
+ * @param value result in case not present
+ * @return its value if present, otherwise [value]
+ */
+fun <T> OptionalInput<T>.orElse(value: T): T {
+    ifPresent {
+        return it
+    }
+    return value
+}
```

Comment:
sadly not really  
this is not the contract they use, and the chance that they want this is basically zero

Reply:
Why do we use something not according to contract then?

Reply:
we had this discussion a while ago  
we decided that we want to disallow null where it is purely optional, to make the user use null only for setting it to null, to prevent confusing null and undefined where it makes a difference

Reply:
whoops, thought this was another code part  
ignore all my comments  
then for the original answer: to trivial, and probably low chance that it is accepted